.github/workflows/action.yml

```diff
@@ -22,11 +22,16 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-carthage-
 
+    - name: Swift Package manager
```

Comment:
I would suggest adding/duplicate another job here. Renaming `tests` to `carthage-tests` and add `spm-tests` to separate the dependency checks and resolve.